File: typescript/base-hmi-types/src/screens/controls/HmiAlarmSortCriterion.ts
import { HmiAlarmColumnType } from "./HmiAlarmColumnType.js";
import { HmiAlarmSortDirection } from "./HmiAlarmSortDirection.js";

export class HmiAlarmSortCriterion {
  column = HmiAlarmColumnType.Unknown;
  sourceColumn?: string;
  direction = HmiAlarmSortDirection.Unknown;
  sourceDirection?: string;
}

File: src/BaseHmiTypes/Screens/Controls/HmiAlarmSortCriterion.cs
namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmSortCriterion
{
    public HmiAlarmColumnType Column { get; set; }

    public string? SourceColumn { get; set; }

    public HmiAlarmSortDirection Direction { get; set; }

    public string? SourceDirection { get; set; }
}
